src/flow/flowstat_mod.F90

```diff
@@ -230,4 +353,556 @@ SUBROUTINE calclpp_grid(kk, jj, ii, lpp, p, bp, dx, dy, dz, &
+        dx, dy, dz, ddx, ddy, ddz, rddx, rddy, rddz)
+
+        ! Subroutine arguments
+        INTEGER(intk), INTENT(in) :: kk, jj, ii
+        REAL(realk), INTENT(inout) :: dfg (kk, jj, ii)
+        REAL(realk), INTENT(in) :: u(kk, jj, ii), v(kk, jj, ii), w(kk, jj, ii)
+        REAL(realk), INTENT(in) :: bp(kk, jj, ii)
+        REAL(realk), INTENT(in) :: gmol
+        REAL(realk), INTENT(in) :: dx(ii), dy(jj), dz(kk)
+        REAL(realk), INTENT(in) :: ddx(ii), ddy(jj), ddz(kk)
+        REAL(realk), INTENT(in) :: rddx(ii), rddy(jj), rddz(kk)
+
+        ! Local Variables
+        INTEGER(intk) :: k, j, i
+        REAL(realk) :: rddxpl, rddypl, rddzpl
+        REAL(realk) :: dxf, dyf, dzf
+        REAL(realk) :: dudx, dudy, dudz, dvdx, dvdy, dvdz, dwdx, dwdy, dwdz
+
+        DO i = 3, ii-2
+            dxf = 0.5*dx(i-1)*rddx(i)
+
+            DO j = 3, jj-2
+                dyf = 0.5*dy(j-1)*rddy(j)
+
+                DO k = 3, kk-2
+                    dzf = 0.5*dz(k-1)*rddz(k)
+
+                    rddxpl = 1.0/(ddx(i) &
+                        + dx(i+1)*(sign(0.25_realk, bp(k, j, i+1)) + 0.25) &
+                        + dx(i-1)*(sign(0.25_realk, bp(k, j, i-1)) + 0.25))
+
+                    rddypl = 1.0/(ddy(j) &
+                        + dy(j+1)*(sign(0.25_realk, bp(k, j+1, i)) + 0.25) &
+                        + dy(j-1)*(sign(0.25_realk, bp(k, j-1, i)) + 0.25))
+
+                    rddzpl = 1.0/(ddz(k) &
+                        + dz(k+1)*(sign(0.25_realk, bp(k+1, j, i)) + 0.25) &
+                        + dz(k-1)*(sign(0.25_realk, bp(k-1, j, i)) + 0.25))
+
+                    dudx = rddx(i)*(u(k, j, i)-u(k, j, i-1))
+
+                    dudy = rddypl * &
+                           ((u(k  ,j+1,i  )-u(k  ,j-1,i  )) * dxf &
+                        +   (u(k  ,j+1,i-1)-u(k  ,j-1,i-1)) * (1.0-dxf))
+
+                    dudz = rddzpl * &
+                           ((u(k+1,j  ,i  )-u(k-1,j  ,i  )) * dxf &
+                        +   (u(k+1,j  ,i-1)-u(k-1,j  ,i-1)) * (1.0-dxf))
+
+                    dvdx = rddxpl * &
+                           ((v(k  ,j  ,i+1)-v(k  ,j  ,i-1)) * dyf &
+                        +   (v(k  ,j-1,i+1)-v(k  ,j-1,i-1)) * (1.0-dyf))
+
+                    dvdy = rddy(i)*(v(k, j, i)-v(k, j-1, i))
+
+                    dvdz = rddzpl * &
+                           ((v(k+1,j  ,i  )-v(k-1,j  ,i  )) * dyf &
+                        +   (v(k+1,j-1,i  )-v(k-1,j-1,i  )) * (1.0-dyf))
+
+                    dwdx = rddxpl * &
+                           ((w(k  ,j  ,i+1)-w(k  ,j  ,i-1)) * dzf &
+                        +   (w(k-1,j  ,i+1)-w(k-1,j  ,i-1)) * (1.0-dzf))
+
+                    dwdy = rddypl * &
+                           ((w(k  ,j+1,i  )-w(k  ,j-1,i  )) * dzf &
+                        +   (w(k-1,j+1,i  )-w(k-1,j-1,i  )) * (1.0-dzf))
+
+                    dwdz = rddz(i)*(w(k, j, i)-w(k-1, j, i))
+
+                    dfg (k, j, i) = gmol * &
```

Comment:
I want to follow up with another comment here after some archeology in the legacy code.
In the old DISSIPGS there were the line `GAKT = (1-IGS)*GMOL + IGS*G(K,J,I)`. The integer flag IGS was passed in as a subroutine argument, and when `DISSIPGS` was called from `SUMSTA` this flag (the last argument) was 1.

This means that the G(k, j, i) was used instead of `gmol` as you use here.

Is this an intentional change from the legacy MGLET `DISSIPGS`?